src/main/java/fr/inria/gforge/spoon/configuration/AbstractSpoonConfigurationBuilder.java

```diff
@@ -115,6 +117,18 @@ public SpoonConfigurationBuilder addSourceClasspath() throws SpoonMavenPluginExc
 		List<String> compileClasspath;
 		try {
 			compileClasspath = project.getCompileClasspathElements();
+			String userHome = System.getProperty("user.home");
+			File mavenHome = new File(userHome + "/.m2/repository");
+			for (Dependency dependency : project.getDependencies()) {
+				String groupPath = dependency.getGroupId().replaceAll("\\.", "/");
+				String artifactName = dependency.getArtifactId();
+				String version = dependency.getVersion();
+				File entry = Paths.get(mavenHome.toString(), groupPath, artifactName, version,
+						artifactName + "-" + version + "." + dependency.getType()).toFile();
```

Comment:
This is also bad since this should be left to maven.

Reply:
I really think this is definitely not the way to go :)

Reply:
going to write a comment later on it.